darts/models/forecasting_model.py

```diff
@@ -254,12 +259,20 @@ def historical_forecasts(self,
                 train_cov = covariates.drop_after(pred_time)
 
             if retrain:
-                if covariates and "covariates" in fit_signature.parameters:
-                    self.fit(series=train, covariates=train_cov)
-                elif covariates and "exog" in fit_signature.parameters:
-                    self.fit(series=train, exog=train_cov)
+                if self.__class__.__bases__[0].__name__ == 'TorchForecastingModel':
+                    if covariates and "covariates" in fit_signature.parameters:
+                        self.fit(series=train, covariates=train_cov, epochs=retrain_epochs)
+                    elif covariates and "exog" in fit_signature.parameters:
+                        self.fit(series=train, exog=train_cov, epochs=retrain_epochs)
+                    else:
+                        self.fit(series=train, epochs=retrain_epochs)
                 else:
-                    self.fit(series=train)
+                    if covariates and "covariates" in fit_signature.parameters:
```

Comment:
At this point in time, this case is also never realized as far as I can tell, since only `TorchForecastingModel` instances expect a `covariates` parameter in their fit function. But it I guess it might be worth leaving it here, especially since if I understand correctly regression models will soon inherit `GlobalForecastingModel`.